### PR TITLE
Feature/deployment

### PR DESCRIPTION
**Pull request by krubenok**
### Tickets:

HCK-35

### List of changes:

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

- *Updated node version to 10.17.0* This is the latest LTS 10.x version and the closest version to what we were using that Heroku supports. the 9.x branch was getting pretty stale so probably time to move on anyways. 
- *Created Heroku Procfile* this will ease deployment onto Heroku and specifies the entry point for running the application to Heroku.
- *Remove Google Cloud Build Scripts* deploying this project to Google Cloud is a holdover when the project had different goals. The project doesn't require the level of complexity that a Kubernetes cluster on Google Cloud provides and as such is no longer needed. 
- *Updated Deployment Documentation* Changed instructions to reflect that Heroku is the target platform. 
- *Renamed npm scripts* `npm run build` was being used in a non-standard way to "build" the API when there was in fact nothing to build. Note for maintainers: `npm run build` is generally expected to terminate and not be a process that remains running. Scripts were renamed to `npm run start` and `npm run start:dev` 
- Deploy to Heroku Button created in README.md

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] New release
- [x] This change requires a documentation update

### How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Tested various calls to new endpoint URL in postman. Seems to work well but please test more thoroughly!! 

**Test Configuration**:
Heroku, Node 10.17.0LTS

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,56 +1,59 @@
...
-    "eslint-plugin-prettier": "3.1.1",
-    "mocha": "^6.2.2",
-    "nodemon": "^1.19.4",
-    "prettier": "1.19.1"
-  }
+    "name": "hackerAPI",
+    "version": "1.6.2",
+    "private": true,
+    "scripts": {
+        "start:dev": "DEBUG=hackboard:* NODE_ENV=development nodemon --ignore gcp_creds.json ./bin/www.js",
```
Comment (loreina) [inline package.json]: can we change your `start:dev` to `start`

File: package.json [hunk 2]
```diff
@@ -1,56 +1,59 @@
...
+        "start": "NODE_ENV=deployment node ./bin/www.js",
```
Comment (loreina) [inline package.json]: can we change your `start` to `deploy`?

File: README.md [hunk 3]
```diff
@@ -15,3 +15,7 @@ API for registration, live-site
 ## How to use and contribute
 
 See documentation here: <https://docs.mchacks.ca>
+
+## Deploy
+
+[![Deploy](https://www.herokucdn.com/deploy/button.svg)](https://heroku.com/deploy)
```
Comment (loreina) [inline README.md]: is the button supposed to link to our Heroku app?

File: docs/deploy.md [hunk 4]
```diff
@@ -1,71 +1,42 @@
...
-    WORKDIR /usr/src/app
-    COPY package*.json ./
-    ADD VERSION .
-    RUN npm install
-    COPY . .
-    EXPOSE 8080
-    CMD [ "npm", "start" ]
+This application can be most easily deployed via Heroku. A "deploy to heroku" button is available in the `README.md` file. Once in Heroku, several environment variables will need to be filled out including: 
+
+```plain
```
Comment (loreina) [inline docs/deploy.md]: will this change our `.env.example`?

Comment (krubenok) [reply docs/deploy.md]: No. Only a subset of that is needed for a production deploy.

File: README.md (hunk 3 above)
Comment (krubenok) [reply README.md]: No. This is a deploy to Heroku button that creates a new instance of the API.

File: package.json [hunk 5]
```diff
@@ -1,56 +1,59 @@
...
+        "apidoc": "^0.17.7",
+        "chai": "^4.2.0",
+        "chai-http": "^4.3.0",
+        "eslint": "6.7.2",
+        "eslint-config-prettier": "6.7.0",
+        "eslint-plugin-prettier": "3.1.1",
+        "mocha": "^6.2.2",
+        "nodemon": "^1.19.4",
+        "prettier": "1.19.1"
+    }
```
Comment (krubenok) [inline package.json]: The VS Code format on save settings forced the 4 tab spacing to be applied to this file hence the large diff.

File: .env.example [hunk 6]
```diff
@@ -6,6 +6,7 @@ ADDRESS=<Some address such as 127.0.0.1>
 #front-end server
 FRONTEND_ADDRESS_DEV=localhost:1337
 FRONTEND_ADDRESS_DEPLOY=app.mchacks.ca
+FRONTEND_ADDRESS_BETA=develop--mchacks-dashboard.netlify.com
```
Comment (loreina) [inline .env.example]: can you remove this for now? I think it's better to discuss w everyone first and we can always add it back easily

**Pull request merged**